newrelic/resource_newrelic_nrql_alert_condition.go

```diff
@@ -85,74 +81,221 @@ func resourceNewRelicNrqlAlertCondition() *schema.Resource {
+							Optional:     true,
+							Description:  "Valid values are: 'all' or 'any'",
 							ValidateFunc: validation.StringInSlice([]string{"all", "any"}, false),
-							Description:  "One of (all, any).",
+						},
+
+						// NerdGraph only. Seems to be similar to `time_function`
+						"threshold_occurrences": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Description:  "Valid values are: 'ALL' or 'AT_LEAST_ONCE'",
+							ValidateFunc: validation.StringInSlice([]string{"ALL", "AT_LEAST_ONCE"}, false),
+						},
+						// NerdGraph only. Equivalent to `duration`, but in seconds
+						"threshold_duration": {
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Description: "The duration of time, in seconds, that the threshold must violate for in order to create a violation. Value must be a multiple of 60 and within 120-3600 seconds for baseline conditions and 120-7200 seconds for static conditions.",
+							ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+								v := val.(int)
+
+								// Value must be a factor of 60.
+								if v%60 != 0 {
+									errs = append(errs, fmt.Errorf("%q must be a factor of 60, got: %d", key, v))
+								}
+
+								// This validation is a top-level validation check.
+								// Baseline conditions must be within range [120, 3600].
+								// Baseline condition validation lives in the "expand" functions.
+								if v < 120 || v > 7200 {
+									errs = append(errs, fmt.Errorf("%q must be between 120 and 7200 inclusive, got: %d", key, v))
+								}
+
+								return
+							},
 						},
 					},
 				},
-				Required: true,
-				MinItems: 1,
 			},
-			"type": {
-				Type:         schema.TypeString,
-				Optional:     true,
-				Default:      "static",
-				ValidateFunc: validation.StringInSlice([]string{"static", "outlier", "baseline"}, false),
+			// Outlier ONLY
+			"expected_groups": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "Number of expected groups when using outlier detection.",
+			},
+			// Outlier ONLY
+			"ignore_overlap": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Whether to look for a convergence of groups when using outlier detection.",
+			},
+			"violation_time_limit_seconds": {
+				Deprecated:    "use `violation_time_limit` attribute instead",
+				Type:          schema.TypeInt,
+				Optional:      true,
+				Description:   "Sets a time limit, in seconds, that will automatically force-close a long-lasting violation after the time limit you select. Possible values are 3600, 7200, 14400, 28800, 43200, and 86400.",
+				ConflictsWith: []string{"violation_time_limit"},
+				ValidateFunc:  validation.IntInSlice([]int{3600, 7200, 14400, 28800, 43200, 86400}),
 			},
+			// Exists in NerdGraph, but with different values. Figure out how to handle this.
+			// Conflicts with `baseline_direction` when using NerdGraph
```

Comment:
This one was weird because it sets a default value of `single_value` and that seemed to cause issues for `ConflictsWith`, but I will double check that. Also open to removing the default value for this, but that seems like a breaking change. But maybe that's okay since this is going in v2.0.0?